lib/assets/javascripts/turbograft/turbolinks.coffee

```diff
@@ -136,49 +136,56 @@ class window.Turbolinks
 
   @loadPage: (url, xhr, options = {}) ->
     triggerEvent 'page:receive'
+    response = new TurboGraft.Response(xhr)
     options.updatePushState ?= true
-    if upstreamDocument = new TurboGraft.Response(xhr).document()
-      if options.partialReplace
-        reflectNewUrl url if options.updatePushState
-        updateBody(upstreamDocument, xhr, options)
-      else
-        turbohead = new TurboGraft.TurboHead(activeDocument, upstreamDocument)
-        if turbohead.hasAssetConflicts()
-          return Turbolinks.fullPageNavigate(url)
-        reflectNewUrl url if options.updatePushState
-        turbohead.waitForAssets().then((result) ->
-          updateBody(upstreamDocument, xhr, options) unless result?.isCanceled
-        )
-    else
+    options.partialReplace = isPartialReplace(options, response)
```

Comment:
It feels gross to assign to an object, using that object as an input into the function (also, options should always come last in a function IMO)